tests/unit/test_charm.py

```diff
@@ -1,25 +1,147 @@
+        setattr(self.harness.charm._stored, "slurm_installed", False)  # Patch StoredState
+        res = self.harness.charm._check_status()
+        self.assertEqual(self.harness.charm.unit.status, BlockedStatus("Error installing slurm"))
+        self.assertEqual(
+            res, False, msg="_check_status returned value True instead of expected value False."
+        )
+
+    def test_check_slurmdbd(self) -> None:
+        """Test that _check_slurmdbd method works."""
+        self.harness.charm._check_slurmdbd(max_attemps=1)
+        self.assertEqual(self.harness.charm._slurm_manager.slurm_is_active(), True)
+
+    @patch("slurm_ops_manager.SlurmManager.slurm_is_active", return_value=False)
+    def test_check_slurmdbd_slurm_not_active(self, _) -> None:
+        """Test that proper block status is thrown if slurm is not active."""
+        self.harness.charm._check_slurmdbd(max_attemps=1)
+        self.assertEqual(self.harness.charm.unit.status, BlockedStatus("Cannot start slurmdbd"))
+
+    def test_on_db_unavailable(self) -> None:
+        """Test the that _on_db_available method works."""
+        self.harness.charm._db.on.database_unavailable.emit()
+        self.assertEqual(self.harness.charm._stored.db_info, {})
+
+    def test_on_jwt_available(self, _) -> None:
+        """Test that _on_jwt_available method works."""
+        setattr(self.harness.charm._stored, "slurm_installed", True)  # Patch StoredState
+        self.harness.charm.on.jwt_available.emit()
+        self.assertEqual(self.harness.charm._stored.jwt_available, True)
+
+    @patch("slurm_ops_manager.SlurmManager.restart_munged", return_value=True)
+    def test_on_munge_available(self, _) -> None:
+        """Test that _on_munge_available method works."""
+        setattr(self.harness.charm._stored, "slurm_installed", True)  # Patch StoredState
+        self.harness.charm.on.munge_available.emit()
+        self.assertEqual(self.harness.charm._stored.munge_available, True)
+
+    @patch("slurm_ops_manager.SlurmManager.restart_munged", return_value=False)
+    def test_on_munge_available_fail_restart(self, _) -> None:
+        """Test that _on_munge_available properly handles when munge fails to restart."""
+        setattr(self.harness.charm._stored, "slurm_installed", True)  # Patch StoredState
+        self.harness.charm.on.munge_available.emit()
+        self.assertEqual(self.harness.charm.unit.status, BlockedStatus("Error restarting munge"))
+
+    def test_on_slurmctld_unavailable(self) -> None:
+        """Test that _on_slurmctld_unavailable method works."""
+        self.harness.charm.on.slurmctld_unavailable.emit()
+        self.assertEqual(self.harness.charm._stored.jwt_available, False)
+        self.assertEqual(self.harness.charm._stored.munge_available, False)
+
+    def test_write_config_and_restart_slurmdbd(self) -> None:
+        """Test that _write_config_and_restart_slurmdbd method works."""
+        self.harness.charm.on.write_config.emit()
```

Comment:
If similar to the upgrade test case just add a comment or update test description about the test failing on catching an error.

Reply:
See https://github.com/canonical/slurmdbd-operator/pull/2#discussion_r1065966052

Reply:
Decided to bomb this unit test out in https://github.com/canonical/slurmdbd-operator/pull/2/commits/6ad37cd848f9a2dcbee3fb1df15b742d19f45260 since this is something that I am planning to change in the charm.